app/models/user.rb

```diff
@@ -28,9 +32,25 @@ class User < ApplicationRecord
   # `uid` is a unique ID that comes back from OmniAuth (which gets it from
   # the remote authentication provider). It is used to lookup or create a new
   # local user via this method.
+  # Touchstone and fake_auth put this UID in different places.
   def self.from_omniauth(auth)
-    where(uid: auth.info.uid).first_or_create do |user|
+    if auth.info.key? 'uid'
+      uid = auth.info.uid
+    else
+      uid = auth.uid
+    end
+
+    User.where(uid: uid).first_or_create do |user|
       user.email = auth.info.email
+      user.given_name = auth.info.given_name
+      user.surname = auth.info.surname
     end
   end
+
+  # Definitely for sure wrong for some people. But staff need name info to
+  # process theses and we have no way of knowing which users would prefer
+  # something else.
```

Comment:
Would it be better to use the displayName attribute we can get from Touchstone that _may_ address some of these concerns? I feel like what we have here is fine, but since you commented it's probably wrong sometimes maybe we should consider using the best data we have?

Reply:
If the whole point is that we're sorting on last name, we need something normalized to last name. If we find ourselves actually showing names to users, maybe we should get displayName. I'll add the comment.